### PR TITLE
#61 Send reset password

### PR DESCRIPTION
**Pull request by greenc123**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const {startServer} = require('./src/server')
+
+startServer(3052)
```
Comment (songz) [inline index.js:3]: `startServer( process.env.PORT || 3052 )` -> somebody may want to run your app on a different port.

File: src/public/hello.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                            <h1>Hello.html</h1>
+
+                                <script>
+                                        fetch('/api/notifications', {
+                                                headers: {
+                                                          Accept: 'application/json',
+                                                          'Content-Type': 'application/x-www-form-urlencoded',
+                                                        },
+                                                method: 'POST',
+                                                body: 'email=alfred@gmail.com&username=Batman',
```
Comment (songz) [inline src/public/hello.html:22]: remove this file

File: src/server.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+const getApp = () => {
+  return app
+}
+
+const startServer = (portNumber) => {
+  return new Promise((resolve, reject) => {
+    app = express()
+
+    app.use(express.static('public'))
```
Comment (songz) [inline src/server.js:16]: don't need this.

File: src/server.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    app.use(express.json()) // for parsing application/json
+    app.use(express.urlencoded({ extended: true })) // for parsing application/x-www-form-urlencoded
```
Comment (songz) [inline src/server.js:18]: let's stick to `json`

File: src/server.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    app.post('/api/notification', async (req, res) => {
+      const email = req.body.email
```
Comment (songz) [inline src/server.js:21]: add a check....
```
if (!email) {
  return res.status(400).json({ error: { message: "invalid input"} })
}
```

File: src/server.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      const {Accounts} = db.getModels()
+      const userAccount = await Accounts.findOne({
+        where: {
+          email: email
+        }
+      })
```
Comment (songz) [inline src/server.js:27]: add a check `if (!userAccount)`

File: src/server.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      try {
+        sendPasswordResetEmail(userAccount) 
+        res.status(200).send('success')
+      } catch (err) {
+       return {
+         error : {
```
Comment (songz) [inline src/server.js:33]: should always send back a response. `res.status(500).json({error: .....})`

**Comment by greenc123** [comment]
This needs to include express router

**Pull request closed without merging**